### PR TITLE
(Source) - Couchbase connector 🎉

### PR DESCRIPTION
**Pull request by teetangh**
## What
* Implementation of a Couchbase source connector that enables data extraction from Couchbase databases
* Supports both full refresh and incremental sync modes
* Handles authentication and connection management with Couchbase clusters
* Implements efficient querying using SQL++ with pagination support

## How
* Uses Python CDK's AbstractSource for the main connector implementation
* Implements connection checking using Couchbase SDK's cluster connection
* Utilizes SQL++ queries with parameterized statements for secure data access
* Handles incremental sync using configurable cursor fields

## Review guide
1. `source.py` - Main connector implementation, connection handling, and stream configuration
2. `streams.py` - Stream implementations for full refresh and incremental sync modes
3. `queries.py` - SQL++ query construction and execution logic
4. Integration tests for verifying connector behavior with actual Couchbase instance
5. Unit tests covering core functionality and edge cases

## User Impact
* Users can reliably sync data from Couchbase databases to any Airbyte destination
* Clear error messages help with troubleshooting
* Incremental sync support reduces data transfer and improves efficiency

## Can this PR be safely reverted and rolled back?
- [x] YES 💚
* The connector is self-contained and doesn't have dependencies on other system components
* No database migrations or schema changes are required
* Reverting would simply disable the Couchbase source functionality without affecting other connectors

### CODE REVIEW

**Comment by deniswsrosa** [comment]
Any updates on this one?

**Comment by marcosmarxm** [comment]
Hello 👋, I apologize if your contribution hasn't been reviewed or updated yet. I'm working to address the community contribution backlog. I created this [GitHub project](https://github.com/orgs/airbytehq/projects/108/views/4) to help you understand when to expect a review. This week, my goal is to complete all contributions related to Marketplace and Manifest-only connectors. Next week, I will focus on API Certified Connectors, followed by DB Java Connector contributions. If you have any questions about your contribution, please feel free to send me a direct message in the Airbyte Community Slack (and include the link to your pull request).

@teetangh make sure to sign the CLA until there.

**Review by marcosmarxm — COMMENTED** [review]
@teetangh the implementation looks good. I'll create a Couchbase cloud account to run integration tests. I hope to get this done next week.

File: airbyte-integrations/connectors/source-couchbase/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          secretStore:
+            type: GSM
+            alias: airbyte-connector-testing-secret-store
+    - suite: acceptanceTests
+      testSecrets:
+        - name: SECRET_SOURCE-COUCHBASE__CREDS
+          fileName: config.json
+          secretStore:
+            type: GSM
+            alias: airbyte-connector-testing-secret-store
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-couchbase/metadata.yaml:44]: @teetangh please comment these lines for now.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-couchbase/metadata.yaml (hunk 1 above)
Comment (teetangh) [reply airbyte-integrations/connectors/source-couchbase/metadata.yaml]: have commented these lines for now

**Review by marcosmarxm — APPROVED** [review]
